addons/common/fnc_isTurnedOut.sqf

```diff
@@ -27,54 +27,53 @@ Examples:
-        _turn = (_forceHideDriver != 1);
-        _out = (_vehicle animationPhase "hatchDriver") > 0;
-    } else {
-        private _assignedRole = assignedVehicleRole _unit;
-        if (_assignedRole select 0 == "Turret") then {
-            private _turretPath = _assignedRole select 1;
-            private _turret = [_vehicle, _turretPath] call CBA_fnc_getTurret;
-
-            private _canHideGunner = getNumber(_turret >> "canHideGunner");
-            private _forceHideGunner = getNumber(_turret >> "forceHideGunner");
-            _turn = (_canHideGunner == 1 && _forceHideGunner == 0);
-
-            private _hatchAnimation = getText(_turret >> "animationSourceHatch");
-            _out = (_vehicle animationPhase _hatchAnimation) > 0;
-
-        } else {
-            if ((_assignedRole select 0) == "Cargo") then {
-                private _attenuateCargo = getArray(_cfg >> "soundAttenuationCargo");
-                if ((count _attenuateCargo) > 0) then {
-                    private _index = (count _attenuateCargo)-1; // wait for command to get cargo index
-
-                    if (_index > -1) then {
-                        if(_index > (count _attenuateCargo)-1) then {
-                            _index = (count _attenuateCargo)-1;
-                        };
-                        if((_attenuateCargo select _index) == 0) then {
-                            _out = true;
-                        };
-                    };
-                };
-            };
-        };
+private _role = _unit call CBA_fnc_vehicleRole;
+
+// FFVs are allways outside
+if ([_unit] call ace_common_fnc_getTurretIndex in ([_vehicle] call ace_common_fnc_getTurretsFFV)) then {_return = true};
+
+// Spezial condition for RHS car windows
+if (_vehicle isKindOf "rhsusf_m1025_w" && {_role in ["driver", "Turret"]}) then {
+    private _cargoIndex = (_vehicle getCargoIndex _unit) min 2;
+    private _window = ['ani_window_1', 'ani_window_2', 'ani_window_4', 'ani_window_3'] select (_cargoIndex + 1);
+    _return = (_vehicle doorPhase _window) > 0;
+};
+if (_return) exitWith {_return}; // return
+
+if (_role == "driver") exitWith {
+    // if _attenuateCargo = [0] the vehicle is all open
+    private _attenuateCargo = getArray (configFile >> "CfgVehicles" >> (typeOf _vehicle) >> "soundAttenuationCargo");
+
+    (_vehicle animationPhase "hatchDriver") > 0 || {_attenuateCargo isEqualTo [0]}; // return
+};
+
+if (_role in ["Turret", "gunner"]) exitWith {
+    private _turret = [_vehicle, _unit call CBA_fnc_turretPath] call CBA_fnc_getTurret;
+
+    private _gunnerAction = getText (_turret >> "gunnerAction");
+    private _hatchAnimation = getText (_turret >> "animationSourceHatch");
+    
+    (_vehicle animationPhase _hatchAnimation) > 0 || {animationState player == _gunnerAction}; // return
+};
+
+if (_role == "cargo") exitWith {
+    private _attenuateCargo = getArray (configFile >> "CfgVehicles" >> (typeOf _vehicle) >> "soundAttenuationCargo");
```

Comment:
You have `configFile >> "CfgVehicles" >> (typeOf _vehicle)` all over the place.
Resolve it once before it'sf irst use. Then store it in a variable and reuse that.

Reply:
driver and cargo need it... but turret, gunner and commander not... so init it every time?

Reply:
why not move gunner/turret above driver/cargo then?

Reply:
like it

Reply:
but the get never executed booth only one of them... so reuse is not there